course/src/isometry.cc

```diff
@@ -0,0 +1,124 @@
+#include "isometry.h"
+#include <cmath>
+#include "matrix3.h"
+#include "vector3.h"
+
+namespace ekumen {
+namespace math {
+namespace {}  // namespace
+
+Isometry::Isometry(const Vector3& translation, const Matrix3& rotation)
+    : translation_(translation), rotation_(rotation) {}
+
+Isometry::Isometry(const Matrix3& rotation)
+    : translation_(Vector3::kZero), rotation_(rotation) {}
+
+Isometry::Isometry(const Isometry& obj)
+    : Isometry(obj.translation_, obj.rotation_) {}
+
+Isometry::Isometry(Isometry&& obj)
+    : translation_(std::move(obj.translation_)),
+      rotation_(std::move(obj.rotation_)) {}
+
+Isometry& Isometry::operator=(const Isometry& obj) {
+  translation_ = obj.translation_;
+  rotation_ = obj.rotation_;
+  return *this;
+}
+Isometry& Isometry::operator=(Isometry&& obj) {
+  // The rvalue reference shouldn't be the same as this.
+  if (this == &obj) {
+    throw std::invalid_argument(
+        "rvalue cannot be identity of lvalue in move assignment.");
+  }
+
+  translation_ = std::move(obj.translation_);
+  rotation_ = std::move(obj.rotation_);
+  return *this;
+}
+
+Isometry Isometry::FromTranslation(const Vector3& translation) {
+  return Isometry(translation);
+}
+
+Isometry Isometry::RotateAround(const Vector3& axis, const double& angle) {
+  Matrix3 res(Matrix3::kIdentity);
+  double cosAngle = std::cos(angle);
+  double sinAngle = std::sin(angle);
+  if (axis == Vector3::kUnitX) {
```

Comment:
This should work regardless of what the axis vector is. It may not be aligned with any normal axis. See [this](https://en.wikipedia.org/wiki/Rotation_matrix#Axis_and_angle).

Reply:
Done, thanks for the link!